src/engine.js

```diff
@@ -9,62 +9,27 @@ const loadTasksFromLocalStorage = () => {
   }
 };
 
-// Save and update local storage
-const saveTasksToLocalStorage = () => {
-  localStorage.setItem('todoList', JSON.stringify(list));
-  const filteredList = list.filter((task) => !task.completed);
-  localStorage.setItem('todoList', JSON.stringify(filteredList));
-};
-
-// Display tasks
-const displayItemsByIndex = () => {
-  listContainer.innerHTML = '';
-
+// Event listeners for the checkboxes and delete buttons
+const attachEventListeners = () => {
   list.forEach((item) => {
-    const { description, completed, index } = item;
-
-    const listItem = document.createElement('div');
-    const descriptionSpan = document.createElement('span');
-    const completedRadio = document.createElement('input');
-    const edit = document.createElement('button');
-    const hr = document.createElement('hr');
-
-    edit.classList.add('edit-task');
-
-    listItem.classList.add('icon');
-
-    descriptionSpan.textContent = description;
-    completedRadio.type = 'checkbox';
-    completedRadio.name = `completed_${index}`;
-    completedRadio.checked = completed;
-    edit.textContent = 'Edit task';
+    const { index } = item;
+    const completedRadio = document.querySelector(`input[name="completed_${index}"]`);
 
-    // Handle checkbox change event
     completedRadio.addEventListener('change', () => {
       item.completed = completedRadio.checked;
       saveTasksToLocalStorage();
     });
```

Comment:
[REQUIRED] Kindly move the code related to status update for completed to a separate new `module`. Take note that it should be imported as a module.